dig.go

```diff
@@ -108,6 +109,22 @@ func Name(name string) ProvideOption {
 	})
 }
 
+// SkipAcyclicVerification is a ProvideOption to override the default behavior
+// of container.Provide to ensure the dependency graph remains acyclic for each
+// new provider.
+//
+// Applications adding providers to a container in a tight loop may experience
+// performance improvements by using this option.
+//
+// NOTE: A dig container has undefined behavior if the dependency graph is not
+// acyclic. It is highly recommended to call container.VerifyAcyclic before
```

Comment:
Any downsides to programatically enforcing this? Something like a hidden verified flag on the container. Then when an invoke is run, if the flag is false we verify the container first.